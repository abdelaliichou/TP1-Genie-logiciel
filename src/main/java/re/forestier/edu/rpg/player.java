package re.forestier.edu.rpg;

import java.util.ArrayList;
import java.util.HashMap;

public class player {
    public String playerName;
    public String Avatar_name;
    private String AvatarClass;

    public Integer money;

    public int healthpoints;
    public int currenthealthpoints;
    protected int xp;

    private final int LEVEL1 = 10;
    private final int LEVEL2 = 27;
    private final int LEVEL3 = 57;
    private final int LEVEL4 = 111;

    protected HashMap<Integer, Integer> levels = new HashMap<>();
    public HashMap<String, Integer> abilities;
    public ArrayList<String> inventory;


    public player(String playerName, String avatar_name, String avatarClass, int money, ArrayList<String> inventory) {
        if (!avatarClass.equals("ARCHER") && !avatarClass.equals("ADVENTURER") && !avatarClass.equals("DWARF") ) {
            return;
        }

        this.playerName = playerName;
        Avatar_name = avatar_name;
        AvatarClass = avatarClass;
        this.money = Integer.valueOf(money);
        this.inventory = inventory;
        this.abilities = UpdatePlayer.abilitiesPerTypeAndLevel().get(AvatarClass).get(1);

        // setting levels her instead of inside the retrieveLevel function
        settingLevels();

    }

    public void removeMoney(int amount) throws IllegalArgumentException {
        if (amount > money) throw new IllegalArgumentException("Player can't have a negative money!");
        money = money - amount;
    }

    public void addMoney(int amount) {
        if (Integer.valueOf(amount) != null) money += amount;
    }

    private void settingLevels(){
        // (lvl-1) * 10 + round((lvl * xplvl-1)/4)
        levels.put(2,10); // 1*10 + ((2*0)/4)
        levels.put(3,27); // 2*10 + ((3*10)/4)
        levels.put(4,57); // 3*10 + ((4*27)/4)
        levels.put(5,111); // 4*10 + ((5*57)/4)
    }

    public int retrieveLevel() {
        return niveau();
    }

    private int niveau(){
        if (xp < LEVEL1) return 1;
        if (xp < LEVEL2) return 2;
        if (xp < LEVEL3) return 3;
        if (xp < LEVEL4) return 4;
        return 5;
    }

    public String getAvatarClass () {
        return AvatarClass;
    }

    public int getXp() {
        return this.xp;
    }

    /*
    Ингредиенты:
        Для теста:

            250 г муки
            125 г сливочного масла (холодное)
            70 г сахара
            1 яйцо
            1 щепотка соли
     */

}