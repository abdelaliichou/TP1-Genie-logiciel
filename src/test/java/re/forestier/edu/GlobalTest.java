package re.forestier.edu;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import re.forestier.edu.rpg.Affichage;
import re.forestier.edu.rpg.UpdatePlayer;
import re.forestier.edu.rpg.player;

import java.util.ArrayList;

import static org.approvaltests.Approvals.verify;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.junit.jupiter.api.Assertions.fail;

public class GlobalTest {

    @Test
    void testAffichageBase() {
        player player = new player("Florian", "Gnognak le Barbare", "ADVENTURER", 200, new ArrayList<>());
        player.currenthealthpoints = player.healthpoints/2 -1 ;
        UpdatePlayer.addXp(player, 20);
        UpdatePlayer.majFinDeTour(player);

        player.currenthealthpoints = player.healthpoints/2 +1;

        player.inventory = new ArrayList<>();

        verify(Affichage.afficherJoueur(player));
    }

    @Test
    void testAffichageBase2() {
        player player = new player("Florian", "Gnognak le Barbare", "ADVENTURER", 200, new ArrayList<>());
        UpdatePlayer.addXp(player, 30);
        UpdatePlayer.addXp(player, 55);
        UpdatePlayer.addXp(player, 70);
        player.getXp();
        player.inventory = new ArrayList<>();

        verify(Affichage.afficherJoueur(player));
    }

    @Test
    void testAffichageBase3() {
        player player = new player("Florian", "Gnognak le Barbare", "ADVENTURER", 200, new ArrayList<>());
        UpdatePlayer.addXp(player, 20);
        player.inventory = new ArrayList<>();
        player.inventory.add("abdelali ichou"); // add element to the list so the for each can be reached in the Affichage class

        verify(Affichage.afficherJoueur(player));
    }
}
