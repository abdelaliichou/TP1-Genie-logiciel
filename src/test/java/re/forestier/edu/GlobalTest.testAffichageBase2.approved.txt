Joueur Gnognak le Barbare joué par Florian
Niveau : 5 (XP totale : 135)

Capacités :
   VIS : 1
   DEF : 4
   ATK : 5
   ALC : 1
   CHA : 3
   INT : 2

Inventaire :